app/src/main/java/com/example/myweatherapp/data/service/WeatherService.kt

```diff
@@ -12,15 +12,17 @@ class WeatherService {
 
     private val api = WeatherRequestGenerator()
 
-    fun getWeekWeather(cityWeather: String): Observable<Weather> {
+    fun getWeekWeather(cityWeather: String): Observable<List<DaysWeather>> {
         return Observable.create { subscriber ->
             val callResponse =
                 api.createService(WeatherApi::class.java)
                     .getForecast(cityWeather, BuildConfig.APPID, UNITS)
             val response = callResponse.execute()
             if (response.isSuccessful) {
-                Log.d(TAG, response.body().toString())
-                (response.body()?.let { subscriber.onNext(it.transform()) })
+                //Log.d(TAG, response.body().toString())
+                response.body()?.list?.filter { it.date.contains(HOUR) }?.let {
```

Comment:
do not make the filter here in the service.. you should do it in the presenter and save all the data in the model without filtered

Reply:
Now i think its done